File: .claude/hooks/task-created-naming-convention.mjs
#!/usr/bin/env node
// @hookstack task-created-naming-convention
// Impose une référence de ticket en tête du sujet d'une tâche (TaskCreated)
import { readFileSync } from 'node:fs';
import { fileURLToPath } from 'node:url';

export function run(input) {
  const subject = input.task_subject ?? '';
  if (/^\[[A-Z]+-\d+\]/.test(subject)) return null;
  return {
    exitCode: 2,
    message:
      `Task subject must start with a ticket reference, e.g. "[PROJ-123] ${subject}". ` +
      'Update the subject to include a valid ticket number.',
  };
}

/* v8 ignore next 6 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
  const input = JSON.parse(readFileSync(0, 'utf8'));
  const result = run(input);
  if (result) {
    process.stderr.write(result.message);
    process.exit(result.exitCode);
  }
}

File: .claude/hooks/task-created-naming-convention.py
#!/usr/bin/env python3
# @hookstack task-created-naming-convention
"""Enforces a ticket reference at the start of a task subject (TaskCreated)."""
import json
import re
import sys

TICKET_RE = re.compile(r"^\[[A-Z]+-\d+\]")


def run(input_data=None):
    subject = (input_data or {}).get("task_subject") or ""
    if TICKET_RE.match(subject):
        return None
    return {
        "exitCode": 2,
        "message": (
            f'Task subject must start with a ticket reference, e.g. "[PROJ-123] {subject}". '
            "Update the subject to include a valid ticket number."
        ),
    }


if __name__ == "__main__":
    data = json.load(sys.stdin)
    result = run(data)
    if result:
        sys.stderr.write(result["message"])
        sys.exit(result["exitCode"])
